components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -53,49 +64,53 @@ open class ExperimentsInternalAPI internal constructor() {
      * as shared preferences.  As we cannot enforce through the compiler that the context pass to
      * the initialize function is a applicationContext, there could potentially be a memory leak
      * if the initializing application doesn't comply.
-     *
      * @param configuration [Configuration] containing information about the experiments endpoint.
+     *
+     * @return Returns a [Job] so that off-thread initialization can be monitored and interacted
+     * with.
      */
     fun initialize(
         applicationContext: Context,
         configuration: Configuration = Configuration(),
         onExperimentsUpdated: (() -> Unit)? = null
-    ) {
+    ): Job? {
```

Comment:
Do we *need* to provide external consumers the ability to monitor the experiments initialization? If yes, why?

If this is only needed in tests, we can createa an `internal` variable from `coroutineScope.launch` and use that for waiting.

Reply:
Good idea! I just pushed this change.